build_tools/circle/build_doc.sh

```diff
@@ -51,5 +52,11 @@ source activate testenv
 # Build and install scikit-learn in dev mode
 python setup.py develop
 
+if [[ "$CIRCLE_BRANCH" =~ ^master$|^[0-9]+\.[0-9]+\.X$ ]]
```

Comment:
Any reason why the -z "$CI_PULL_REQUEST" has disappeared from my latest comment [here](https://github.com/scikit-learn/scikit-learn/pull/7671#issuecomment-255779852).

> My understanding is that we should never build the pdf in pull requests because it takes too much time

My expectation is that CIRCLE_BRANCH is going to be master in a PR (because the PR is targeting master). I may be wrong about that.